bin/utils/subkey/README.md

```diff
@@ -1 +1,192 @@
+  Public key (hex): 0xfec70cfbf1977c6965b5af10a4534a6a35d548eb14580594d0bc543286892515
+  Account ID:       0xfec70cfbf1977c6965b5af10a4534a6a35d548eb14580594d0bc543286892515
+  SS58 Address:     5Hpm9fq3W3dQgwWpAwDS2ZHKAdnk86QRCu7iX4GnmDxycrte
+```
+### Signing
+
+`subkey` allows using a **secret key** to sign a random message. The signature can then be verified by anyone using your **public key**:
+
+    echo -n <msg> | subkey sign --suri <seed,mnemonic>
+
+*Example*:
+
+    MESSAGE=hello
+    SURI=0xa05c75731970cc7868a2fb7cb577353cd5b31f62dccced92c441acd8fee0c92d
+    echo -n $MESSAGE | subkey sign --suri $SURI
+
+*outputs*:
+
+    9201af3788ad4f986b800853c79da47155f2e08fde2070d866be4c27ab060466fea0623dc2b51f4392f4c61f25381a62848dd66c5d8217fae3858e469ebd668c
+
+**NOTE**: Each run of the `sign` command with yield a different output. While each signature is different, they are all valid.
+
+### Verifying a signature
+
+Given a message, a signature and an address, `subkey` can verify whether the **message** has been digitally signed by the holder (or one of the holders) of the **private key** for the given **address**:
+
+    echo -n <msg> | subkey verify <sig> <address>
+
+*Example*:
+
+    MESSAGE=hello
+    URI=0xfec70cfbf1977c6965b5af10a4534a6a35d548eb14580594d0bc543286892515
+    SIGNATURE=9201af3788ad4f986b800853c79da47155f2e08fde2070d866be4c27ab060466fea0623dc2b51f4392f4c61f25381a62848dd66c5d8217fae3858e469ebd668c
+    echo -n $MESSAGE | subkey verify $SIGNATURE $URI
+
+*outputs*:
+
+    Signature verifies correctly.
+
+A failure looks like:
+
+    Error: SignatureInvalid
+
+### Using the vanity generator
+
+You can use the included vanity generator to find a seed that provides an address which includes the desired pattern. Be warned, depending on your hardware this may take a while.
+
+command:
+```
+subkey vanity --network polkadot --pattern bob
+```
+
+output:
+```
+Generating key containing pattern 'bob'
+best: 190 == top: 189
+Secret Key URI `0x8c9a73097f235b84021a446bc2826a00c690ea0be3e0d81a84931cb4146d6691` is account:
+  Secret seed:      0x8c9a73097f235b84021a446bc2826a00c690ea0be3e0d81a84931cb4146d6691
+  Public key (hex): 0x1a8b32e95c1f571118ea0b84801264c3c70f823e320d099e5de31b9b1f18f843
+  Account ID:       0x1a8b32e95c1f571118ea0b84801264c3c70f823e320d099e5de31b9b1f18f843
+  SS58 Address:     1bobYxBPjZWRPbVo35aSwci1u5Zmq8P6J2jpa4kkudBZMqE
+```
+
+`Bob` now got a nice address starting with his name: 1**bob**YxBPjZWRPbVo35aSwci1u5Zmq8P6J2jpa4kkudBZMqE.
+
+**Note**: While `Bob`, having a short name (3 chars), got a result rather quickly, it will take much longer for `Alice` who has a much longer name, thus the chances to generate a random address that contains the chain `alice` will be much smaller.
+
+### Signing a transaction
+
+Sign a transaction from an encoded `Call`.
+
+    subkey sign-transaction \
+        --call <call-as-hex> \
+        --nonce 0 \
+        --suri <secret-uri> \
+        --password <password> \
```

Comment:
There's been no previous mention of the password in the doc, maybe worth including some info on how that works?